src/Controls/src/Core/IImageSourceHttpClientFactory.cs

```diff
@@ -0,0 +1,12 @@
+#nullable enable
+
+using System;
+using System.Net.Http;
+
+namespace Microsoft.Maui.Controls;
+
+public interface IImageSourceHttpClientFactory
+{
+	bool ShouldDispose { get; }
```

Comment:
What is the purpose of this? When would you set true, and when false?

Reply:
It also seems odd that this setting is for _all_ HttpClients created by this service, as opposed to _particular_ HttpClients created by it.

Reply:
HttpClient is actually an expensive resource and should be reused, or at least the parts that constitute the HttpClient should be reused. At the same time one needs to do cleanup over time. See https://www.stevejgordon.co.uk/httpclient-creation-and-disposal-internals-should-i-dispose-of-httpclient

ShouldDispose just tells if the factory will manage resources automatically or not.

Reply:
The fact that it's done for all HttpClients is okay. It's really just tied to logic of the HttpClient factory which should stay the same for the duration of the app.

Reply:
Naming is hard, perhaps `ShouldDispose` is not the best one :)

Reply:
I would guess the following code will become a standard pattern for people using the `IImageSourceHttpClientFactory` interface. The reason being that `Microsoft.Extensions.Http.IHttpClientFactory` does a lot of the heavy lifting and integrates nicely with Polly. That could be the default, but at the same time you wouldn't want all MAUI programmers to pay the price of including `Microsoft.Extensions.Http`.

```
sealed class ImageSourceHttpClientFactory : IImageSourceHttpClientFactory
{
	readonly IHttpClientFactory _httpClientFactory;

	public ImageSourceHttpClientFactory(IHttpClientFactory httpClientFactory)
	{
		_httpClientFactory = httpClientFactory;
	}

	public bool ShouldDispose => false;

	public HttpClient CreateClient(Uri _)
	{
		// NOTE: possible to add logic depending upon image uri
		return _httpClientFactory.CreateClient(DefaultHttpClientName);
	}

	public static string DefaultHttpClientName => nameof(ImageSourceHttpClientFactory);
}
```

Reply:
Indeed, reusing HttpClient can be hugely important. So then when _would_ you want it disposed? Or is that more to maintain the original behavior where by default it was created and disposed each time? If that's the only purpose, then I would just change the behavior to never dispose and not even give an option.

Reply:
We'll. The default is to create an new HttpClient and that would need to be disposed.